Need to package both static and dynamic libraries for components

Currently only dynamic libraries are being built. This causes problems in projects with build dependencies on boost requiring the static libraries:

https://circleci.com/gh/conda-forge/staged-recipes/3253

```
CMake Error at /opt/conda/envs/_build/share/cmake-3.5/Modules/FindBoost.cmake:1647 (message):
  Unable to find the requested Boost libraries.

  Boost version: 1.60.0

  Boost include path: /opt/conda/envs/_build/include

  Could not find the following static Boost libraries:

          boost_filesystem
          boost_thread
          boost_unit_test_framework
          boost_date_time
          boost_atomic
```

If no one beats me to it, here is one example of building the complete Boost library stack:

https://github.com/cloudera/native-toolchain/blob/master/source/boost/build.sh

--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -26,7 +26,7 @@ if errorlevel 1 exit 1
     address-model=%ARCH% ^
     variant=release ^
     threading=multi ^
-    link=shared ^
+    link=static,shared ^
     -j%CPU_COUNT% ^
     -s ZLIB_INCLUDE="%LIBRARY_INC%" ^
     -s ZLIB_LIBPATH="%LIBRARY_LIB%"

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -33,7 +33,7 @@ if [ "$(uname)" == "Darwin" ]; then
         architecture=x86 \
         debug-symbols=off \
         threading=multi \
-        link=shared \
+        link=static,shared \
         toolset=clang \
         python="${PY_VER}" \
         include="${INCLUDE_PATH}" \
@@ -61,7 +61,7 @@ if [ "$(uname)" == "Linux" ]; then
         debug-symbols=off \
         threading=multi \
         runtime-link=shared \
-        link=shared \
+        link=static,shared \
         toolset=gcc \
         python="${PY_VER}" \
         include="${INCLUDE_PATH}" \

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -11,7 +11,7 @@ source:
   md5: 65a840e1a0b13a558ff19eeb2c4f0cbe
 
 build:
-  number: 1
+  number: 2
   features:
     - vc9               # [win and py27]
     - vc10              # [win and py34]
@@ -37,6 +37,11 @@ test:
     - test -f $PREFIX/lib/libboost_system.a                   # [unix]
     - test -f $PREFIX/lib/libboost_test_exec_monitor.a        # [unix]
     - test -f $PREFIX/lib/libboost_timer.a                    # [unix]
+    - test -f $PREFIX/lib/libboost_filesystem.a               # [unix]
+    - test -f $PREFIX/lib/libboost_thread.a                   # [unix]
+    - test -f $PREFIX/lib/libboost_unit_test_framework.a      # [unix]
+    - test -f $PREFIX/lib/libboost_date_time.a                # [unix]
+    - test -f $PREFIX/lib/libboost_atomic.a                   # [unix]
 
     # Verify dynamic libraries.
     {% set win_vstr = "_".join(version.split(".")[:2]) %}

